eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\MVC\Symfony\Controller\Content;
+
+use eZ\Bundle\EzPublishIOBundle\BinaryStreamResponse;
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\Core\IO\IOService;
+use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class DownloadController extends Controller
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    /** @var \eZ\Publish\Core\IO\IOService */
+    private $ioService;
+
+    public function __construct( ContentService $contentService, IOService $ioService )
+    {
+        $this->contentService = $contentService;
+        $this->ioService = $ioService;
+    }
+
+    public function downloadBinaryFileAction( $contentId, $fieldIdentifier, $downloadName, Request $request )
+    {
+        $content = $this->contentService->loadContent( $contentId );
+
+        $language = $request->query->has( 'language' ) ? $request->query->get( 'language' ) : null;
+        $fieldValue = $content->getFieldValue( $fieldIdentifier, $language );
```

Comment:
Here as well, please use TranslationHelper like for [`ez_field_value()` Twig function](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php#L140), as it [loops against available languages to return the right field](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Helper/TranslationHelper.php#L112-L123).